azext_connect/custom.py

```diff
@@ -382,10 +383,49 @@ def create_resource(service, resource_group, deployment_id, settings, para_dict)
         spring_cloud_handler(resource_group, deployment_id, settings, para_dict)
 
 
+COSMOSDB_KIND = ['GlobalDocumentDB', 'MongoDB', 'Parse']
+COSMOS_CAPABILITES = ['EnableCassandra', 'EnableTable', 'EnableGremlin']
+COSMOS_DATABASE_TYPE = ['cassandraKeyspaces', 'tables', 'gremlinDatabases', 'sqlDatabases', 'mongodbDatabases']
+
+
 def _is_resourcid(resource):
     return resource.startswith('/subscriptions/')
 
 
+def _get_rg_from_scope(scope):
+    if scope.startswith('/subscriptions'):
+        match = re.search(r"\/resourceGroups\/[^\/]+", scope)
+        if (match):
+            return match.group().split("/")[2]
+    return None
+
+
+def _get_cosmos_database_type(resource_group, cosmos_account):
+    cli = get_default_cli()
+    parameters = [
+        'cosmosdb', 'show',
+        '--name', cosmos_account,
+        '--resource-group', resource_group,
+        '--output', 'none'
+    ]
+    if cli.invoke(parameters):
+        raise CLIError('Fail to show CosmosDb account %s info.' % cosmos_account)
+    kind = cli.result.result['kind']
+    capabilites = cli.result.result['capabilities']
+    if kind == COSMOSDB_KIND[0]:
+        for item in capabilites:
+            if item.Name == COSMOS_CAPABILITES[0]:
+                return COSMOS_DATABASE_TYPE[0]
+            if item.Name == COSMOS_CAPABILITES[1]:
+                return COSMOS_DATABASE_TYPE[1]
+            if item.Name == COSMOS_CAPABILITES[2]:
+                return COSMOS_DATABASE_TYPE[2]
+            return COSMOS_DATABASE_TYPE[3]
+    if kind == COSMOSDB_KIND[1]:
+        return COSMOS_DATABASE_TYPE[4]
+    return 'unknown'
```

Comment:
Seems we don't handle the case of "unknown" database type in the caller

Reply:
I changed to throw exception here.